src/SFA.DAS.Campaign.Web/Helpers/SessionService.cs

```diff
@@ -0,0 +1,32 @@
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace SFA.DAS.Campaign.Web.Helpers
+{
+    public interface ISessionService
+    {
+        T Get<T>(string key);
+        void Set<T>(string key, T toStore);
+    }
+
+    public class SessionService : ISessionService
+    {
+        private readonly IHttpContextAccessor _contextAccessor;
+        public SessionService(IHttpContextAccessor contextAccessor)
+        {
+            _contextAccessor = contextAccessor;
+        }
+        
+        public T Get<T>(string key)
+        {
+            return _contextAccessor.HttpContext.Session.GetString(key) != null 
```

Comment:
Should this have a try catch around it for the JsonException that will be thrown if the string is not JSON?
This class is not the only way to get items into the Session of the Context so there is the potential for a bug here and you'd want to at least log that out.

Reply:
I've added some logging to this and changed it so that it's not _every_ exception it is swallowing.